src/services/mpc/recovery/SmsOtpRecovery.ts

```diff
@@ -0,0 +1,292 @@
+import { toPrivKeyEC } from '@tkey-mpc/common-types'
+import {
+  BrowserStorage,
+  FactorKeyTypeShareDescription,
+  generateFactorKey,
+  type MPCKeyDetails,
+  TssShareType,
+  type Web3AuthMPCCoreKit,
+} from '@web3auth/mpc-core-kit'
+import { BN } from 'bn.js'
+import { ecsign, keccak256 } from 'ethereumjs-util'
+import { hexZeroPad } from 'ethers/lib/utils'
+import { SOCIAL_WALLET_OPTIONS } from '../config'
+import EncryptionUtil from '../EncryptionUtil'
+
+type SmsOtpStoreData = {
+  number: string
+  moduleName: string
+}
+
+interface IRegisterRequestBody {
+  pubKey: {
+    x: string
+    y: string
+  }
+  sig: {
+    r: string
+    s: string
+    v: string
+  }
+  identifier: string
+}
+
+interface IStartRequestBody {
+  // `${pubKey.x.toString(16, 64)}{pubkey.y.toString(16, 64)}`
+  address: string
+
+  // web3auth client id.
+  client_id: string
+
+  // This is the value you get from the first /start response body.
+  // Save this in session storage.
+  // Send this value in next /start requests (page reload, resend code, verify).
+  // It helps the backend to identify a unique verification session.
+  tracking_id?: string
+}
+
+interface IVerifyRequestBody {
+  // `${pubKey.x.toString(16, 64)}{pubkey.y.toString(16, 64)}`
+  address: string
+
+  // web3auth client id.
+  client_id: string
+
+  // Get it from the session storage, saved previously from the /start request.
+  tracking_id: string
+
+  // unique identifier, entered by the user
+  code: string
+
+  // custom metadata you want to set with the user.
+  // this is only for the first time, when user is setting up the MFA for the first time.
+  // Send the factorKey in this, which could be in raw format or encrypted using the postbox key.
+  // This can always be updated in the future as well.
+  data?: string
+}
+
+export const SMS_OTP_MODULE_NAME = 'SmsRecovery'
```

Comment:
Pls change this value to `external_sms`, this is to keep it in sync with the PnP version